dockertest/docker_daemon.py

```diff
@@ -159,16 +159,32 @@ def pid():
     if not stdout.startswith('MainPID='):
         raise RuntimeError("Unexpected output from %s: expected MainPID=NNN,"
                            " got '%s'" % (cmd_result.command, stdout))
-    return int(stdout[8:])
-
-
-def cmdline():
+    mainpid = int(stdout[8:])
+    cmd = cmdline(mainpid)
+    if 'dockerd' in cmd[0]:
+        return mainpid
+    # As of April 2017 we may be running dockerd under runc. If so, actual
+    # dockerd process is an immediate child of the one returned by systemd.
+    for child_pid in utils.run("pgrep -P %s" % mainpid).stdout.split():
```

Comment:
Eek. `utils.get_children()` includes threads, which I'm pretty sure is not the desired behavior. `pgrep` exists on RHEL Atomic 7.3, which is where I'm doing all this docker-as-container testing.

Under what circumstance will systemd return a PID that is not alive? I put in the 'Unexpected output' check above because I can envision systemd returning an empty string, or some sort of warning, but I can't imagine it returning MainPID=123 where 123 is invalid. What situation are you envisioning?